### PR TITLE
Add addGamePlayerActionAt

### PR DESCRIPTION
**Pull request by ryotogashi**

### CODE REVIEW

**Comment by ryotogashi** [comment]
@axross Can you check when you have time?

File: hooks/usePlayerActions/addGameStreetActionAt.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  const newActions = [...actions];
+
+  if (index < newActions.length) {
+    newActions.splice(index, 0, action);
+  }
+
+  if (index === newActions.length) {
+    newActions.push(action);
+  }
```
Comment (axross) [inline hooks/usePlayerActions/addGameStreetActionAt.ts:25]:
```suggestion
  newActions.splice(index, 0, action);
```

File: hooks/usePlayerActions/addGameStreetActionAt.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  actions: GameStreetAction[];
+  index: number;
+  action: GameStreetAction;
+}): GameStreetAction[] {
+  if (index < 0 || actions.length < index)
+    throw new Error(
+      "index must be more than or equal to 0 and less than or equal to the length of the street"
+    );
+
+  const newActions = [...actions];
```
Comment (axross) [inline hooks/usePlayerActions/addGameStreetActionAt.ts:17]: [IMO]

```suggestion
  const nextActions = [...actions];
```

File: hooks/usePlayerActions/addGameStreetActionAt.test.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      },
+    });
+
+    expect(actions).toEqual([
+      {
+        type: GameStreetActionType.fold,
+        playerIndex: 0,
+        betSize: 0,
+      },
+    ]);
```
Comment (axross) [inline hooks/usePlayerActions/addGameStreetActionAt.test.ts:58]:
```suggestion
    expect(addGameStreetActionAt({
      actions: [],
      index: 0,
      action: {
        type: GameStreetActionType.fold,
        playerIndex: 0,
        betSize: 0,
      },
    })).toEqual([
      {
        type: GameStreetActionType.fold,
        playerIndex: 0,
        betSize: 0,
      },
    ]);
```

Comment (axross) [reply hooks/usePlayerActions/addGameStreetActionAt.test.ts:58]: Or

```suggestion
    const original = [];
    const added = addGameStreetActionAt({
      actions: original,
      index: 0,
      action: {
        type: GameStreetActionType.fold,
        playerIndex: 0,
        betSize: 0,
      },
    });

    expect(added).toEqual([
      {
        type: GameStreetActionType.fold,
        playerIndex: 0,
        betSize: 0,
      },
    ]);
```

**Pull request merged**